src/main/scala/com/databricks/spark/redshift/Utils.scala

```diff
@@ -47,4 +55,38 @@ private[redshift] object Utils {
    */
   def makeTempPath(tempRoot: String): String = Utils.joinUrls(tempRoot, UUID.randomUUID().toString)
 
+  /**
+   * Checks whether the S3 bucket for the given UI has an object lifecycle configuration to
+   * ensure cleanup of temporary files. If no applicable configuration is found, this method logs
+   * a helpful warning for the user.
+   */
+  def checkThatBucketHasObjectLifecycleConfiguration(
+      tempDir: String,
+      awsCredentials: AWSCredentials): Unit = {
+    try {
+      val s3Client = new AmazonS3Client(
+        new BasicAWSCredentials(awsCredentials.accessKey, awsCredentials.secretAccessKey))
```

Comment:
Argh, potential bug here: with EC2 IAM credentials we won't have a key here, so `awsCredentials` really needs to be an `Option` here (or we need to change the interface).